services/records/records/parsers.go

```diff
@@ -0,0 +1,75 @@
+package records
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/leanspace/terraform-provider-leanspace/helper"
+	"github.com/leanspace/terraform-provider-leanspace/helper/general_objects"
+)
+
+func (record *Record) ToMap() map[string]any {
+	resourceMap := make(map[string]any)
+	resourceMap["id"] = record.ID
+	resourceMap["record_template_id"] = record.RecordTemplateId
+	resourceMap["name"] = record.Name
+	resourceMap["record_state"] = record.RecordState
+	resourceMap["start_date_time"] = record.StartDateTime
+	resourceMap["stop_date_time"] = record.StopDateTime
+	if record.Properties != nil {
+		resourceMap["properties"] = helper.ParseToMaps(record.Properties)
+	}
+	resourceMap["tags"] = helper.ParseToMaps(record.Tags)
+	resourceMap["comment"] = record.Comment
```

Comment:
it seems you always get a comments array from the api so you will need to do some modification here